lib/active_model/serializer/adapter/json_api/link.rb

```diff
@@ -3,30 +3,43 @@ class Serializer
     module Adapter
       class JsonApi
         class Link
-          def initialize(serializer)
+          def initialize(serializer, value)
             @object = serializer.object
             @scope = serializer.scope
+            if value.respond_to?(:call)
+              self._value = instance_eval(&value)
+            else
+              self._value = value
```

Comment:
What's our desired behavior if someone does

```ruby
link :foo do
  meta 'thing'
  'https://example.com/doohickie
end
```

?

right now it would only return the block's return value